py/jpy/src/main/java/org/jpy/PyObjectReferences.java

```diff
@@ -81,33 +81,30 @@ public int cleanupOnlyUseFromGIL() {
     }
 
     private int cleanupOnlyUseFromGIL(long[] buffer) {
-        if (!PyLib.hasGil()) {
-            throw new IllegalStateException(
-                "We should only be calling PyObjectReferences.cleanupOnlyUseFromGIL if we have the GIL!");
-        }
-
-        int index = 0;
-        while (index < buffer.length) {
-            final Reference<? extends PyObject> reference = referenceQueue.poll();
-            if (reference == null) {
-                break;
+        return PyLib.ensureGil(() -> {
```

Comment:
Given this new construction, it likely makes sense to remove the Proxy-via-python than was previously enforcing the GIL; I'm happy to take lead on that part if you like.

Reply:
The proxy no-op may still serve a purpose, though it marginally more expensive than just writing more C, it is less complex to write - we're still not sure about the cost of making a new PyState per formula call. I have a followup patch I'm trying to benchmark using basically the same thing as the old proxy, but the runs I've done are very inconsistent., more than 25% differences in test runs. I'll sync up with you about that, it might well be a better project for you to consider.